### ISSUE TITLE
Jenkins Checkstyle Badge

### ISSUE THREAD

**Issue by fabianhintringer** [badges/shields#6163]
:clipboard: **Description**

We are currently using GitHub and Jenkins as our tools. We would like to ask if it's possible to create a badge to show the Checkstyle output we get on our Jenkins instance.

:link: **Data**

:microphone: **Motivation**

**Comment by calebcartwright** [comment]
Hi :wave: thanks for reaching out! Could you help us out a bit by answering some of the questions from the issue template to provide more context (e.g. what would the badge display/what information is it showing, where can that data be retrieved from)?

It's tough to answer the question about the feasibility of a badge without knowing what exactly you had in mind for the badge message, but generally speaking, if the service/tool/api makes requisite data points available then there's certainly the _potential_ for a badge.

**Comment by fabianhintringer** [comment]
Hello 👋 
# Badge Request
We use the Gradle plugin Checkstyle and have a [Jenkins plugin](https://plugins.jenkins.io/warnings-ng/) installed called "Warnings Next Generation".
The API Endpoint is (there are xml/json endpoints, preferably json) : 
https://jenkinsInstanceUrl.com/job/project_name/job/branch_name/lastBuild/warnings-ng/api/json
https://jenkinsInstanceUrl.com/job/project_name/job/branch_name/lastBuild/warnings-ng/api/xml

The badge should at least contain the "size" field which is the amount of checkstyle complaints.
The label could be "Checkstyle".

# A question
But that's not all:
 - We have our self hosted instance of the repo
 - Created a local.yml where we defined our jenkins url, jenkins_user and jenkins_pass
 - Other badges are working fine and get displayed correctly 🙏 

We know want to know if there is a way to create a dynamic badge which will display us the data of the API endpoint AND authenticates itself.
We already tried:
https://shieldsInstance.com/badge/dynamic/xml?label=checkstyle&query=aggregationApi%2Ftool%2Fsize&url="URL"
where 
"URL" = https://username:password@APIEndpointUrl.com
which was working perfectly, but we do not want our username and password displayed in the URL.


## *Image belongs to "Badge Request"
![Checkstyle_API](https://user-images.githubusercontent.com/66937832/107773173-92e75980-6d3d-11eb-8ed0-a36555f06d4e.png)

**Comment by chris48s** [comment]
If anyone is going to work on this we need to find an example of a public jenkins project where we can call this endpoint (for dev and test purposes). There are some big public instances we use for the other jenkins badges like

- https://wso2.org/jenkins/
- https://ci.eclipse.org/
- https://jenkins.sqlalchemy.org/

**Comment by calebcartwright** [comment]
> We know want to know if there is a way to create a dynamic badge which will display us the data of the API endpoint AND authenticates itself.

No, and I don't really see a feasible way in which that could be done given the inherent open-ended nature of the dynamic badges. If we're able to add a native jenkins checkstyle then it will be able to leverage the same jenkins creds as the other jenkins badges though
